apps/entity-service/src/entities/file-upload.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Controller, Post, Param, NotFoundException, Req } from "@nestjs/common";
+import { FileUploadService } from "../trees/file-upload.service";
+import { EntitiesService } from "./entities.service";
+import { EntityModel } from "@terramatch-microservices/database/constants/entities";
+import { SpecificEntityDto } from "./dto/specific-entity.dto";
+import { PolicyService } from "@terramatch-microservices/common/policies/policy.service";
+import { FastifyRequest } from "fastify";
+
+@Controller("entities/v3/file/upload")
```

Comment:
Two things:
* Please use plural for the base path name
* `upload` is redundant. If you have an API endpoint that accepts a post with media on it, it's pretty clear that's an upload.

So, I'd use `@Controller("entities/v3/files")` for this. In the `app.module.ts`, you'll need to move this new controller ahead of `EntitiesController` in the list of controllers to avoid getting swallowed up by the wildcard routes in that controller (see my in-progress tasks branch: https://github.com/wri/terramatch-microservices/blob/dc8e2a82c1c5ad93717cda01aa70ce4e4900ff78/apps/entity-service/src/app.module.ts#L15-L18)

Reply:
changed